evennia/utils/funcparser.py

```diff
@@ -1252,6 +1252,76 @@ def funcparser_callable_you_capitalize(
     )
 
 
+def funcparser_callable_your(
+    *args, caller=None, receiver=None, mapping=None, capitalize=False, **kwargs
+):
+    """
+    Usage: $your() or $your(key)
+
+    Replaces with your for the caller of the string, with the display_name +'s
+    of the caller for others.
+
+    Keyword Args:
+        caller (Object): The 'your' in the string. This is used unless another
+            your-key is passed to the callable in combination with `mapping`.
+        receiver (Object): The recipient of the string.
+        mapping (dict, optional): This is a mapping `{key:Object, ...}` and is
+            used to find which object `$you(key)` refers to. If not given, the
+            `caller` kwarg is used.
+        capitalize (bool): Passed by the You helper, to capitalize you.
+
+    Returns:
+        str: The parsed string.
+
+    Raises:
+        ParsingError: If `caller` and `receiver` were not supplied.
+
+    Notes:
+        The kwargs should be passed the to parser directly.
+
+    Examples:
+        This can be used by the say or emote hooks to pass actor stance
+        strings.
+
+        - `$your() pet jumps at $you(tommy).`
+
+        The caller-object will see "Your pet jumps Tommy."
+        Tommy will see "CharName's pet jumps at you."
+        Others will see "CharName's pet jumps at Tommy."
+
+    """
+    if args and mapping:
+        # this would mean a $your(key) form
+        try:
+            caller = mapping.get(args[0])
+        except KeyError:
+            pass
+
+    if not (caller and receiver):
+        raise ParsingError("No caller or receiver supplied to $your callable.")
+
+    capitalize = bool(capitalize)
+    if caller == receiver:
+        return "Your" if capitalize else "your"
+
+    name = caller.get_display_name(looker=receiver) \
+        if hasattr(caller, "get_display_name") else str(caller)
+
+    return name + "'s"
```

Comment:
I was taught that if the name ends with `s`, it would be just a single apostrophe, like `Charles'` . But looking it up, it appears it's allowed to also use `'s` here, like `Charles's`. English is weird.

Reply:
I may not be using the exactly correct grammatical terms, but `Charles'` would be a multiple possessive for a word ending in s while `Charles's` is singular possessive or... something like that.

The intent of this function is to show...

"Your ability level has increased!" to Charles and
"Charles's ability level has increased!" to everyone watching.

So it should be correct as-is.

Multiple possessive would be...

"Charles' abilities have increased in level!" and that's not possible to rectify using funcparser.

Reply:
`Charles'` is the correct singular possessive firm of `Charles`

Reply:
O_o
If you say so...
That's not how I was taught in school.